pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go

```diff
@@ -18,99 +18,287 @@ package volumebinding
 			node: &v1.Node{},
-			volumeBinderConfig: &scheduling.FakeVolumeBinderConfig{
-				FindReasons: []scheduling.ConflictReason{scheduling.ErrReasonBindConflict},
+			pvcs: []*v1.PersistentVolumeClaim{
+				makePVC("pvc-a", "", waitSC.Name),
 			},
-			wantStatus: framework.NewStatus(framework.UnschedulableAndUnresolvable, string(scheduling.ErrReasonBindConflict)),
+			wantStateAfterPreFilter: &stateData{
+				boundClaims: []*v1.PersistentVolumeClaim{},
+				claimsToBind: []*v1.PersistentVolumeClaim{
+					makePVC("pvc-a", "", waitSC.Name),
+				},
+			},
+			wantFilterStatus: framework.NewStatus(framework.UnschedulableAndUnresolvable, string(scheduling.ErrReasonBindConflict)),
 		},
 		{
 			name: "bound and unbound unsatisfied",
-			pod:  &v1.Pod{Spec: volState},
-			node: &v1.Node{},
-			volumeBinderConfig: &scheduling.FakeVolumeBinderConfig{
-				FindReasons: []scheduling.ConflictReason{scheduling.ErrReasonBindConflict, scheduling.ErrReasonNodeConflict},
+			pod:  makePod("pod-a", []string{"pvc-a", "pvc-b"}),
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"foo": "barbar",
+					},
+				},
 			},
-			wantStatus: framework.NewStatus(framework.UnschedulableAndUnresolvable, string(scheduling.ErrReasonBindConflict), string(scheduling.ErrReasonNodeConflict)),
+			pvcs: []*v1.PersistentVolumeClaim{
+				makePVC("pvc-a", "pv-a", waitSC.Name),
+				makePVC("pvc-b", "", waitSC.Name),
+			},
+			pvs: []*v1.PersistentVolume{
+				addPVNodeAffinity(makePV("pv-a"), &v1.VolumeNodeAffinity{
+					Required: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{
+									{
+										Key:      "foo",
+										Operator: v1.NodeSelectorOpIn,
+										Values:   []string{"bar"},
+									},
+								},
+							},
+						},
+					},
+				}),
+			},
+			wantStateAfterPreFilter: &stateData{
+				boundClaims: []*v1.PersistentVolumeClaim{
+					makePVC("pvc-a", "pv-a", waitSC.Name),
+				},
+				claimsToBind: []*v1.PersistentVolumeClaim{
+					makePVC("pvc-b", "", waitSC.Name),
+				},
+			},
+			wantFilterStatus: framework.NewStatus(framework.UnschedulableAndUnresolvable, string(scheduling.ErrReasonNodeConflict), string(scheduling.ErrReasonBindConflict)),
 		},
 		{
-			name:               "unbound/found matches/bind succeeds",
-			pod:                &v1.Pod{Spec: volState},
-			node:               &v1.Node{},
-			volumeBinderConfig: &scheduling.FakeVolumeBinderConfig{},
-			wantStatus:         nil,
+			name:                "pvc not found",
+			pod:                 makePod("pod-a", []string{"pvc-a"}),
+			node:                &v1.Node{},
+			wantPreFilterStatus: framework.NewStatus(framework.Error, `error getting PVC "default/pvc-a": could not find v1.PersistentVolumeClaim "default/pvc-a"`),
+			wantFilterStatus:    nil,
 		},
 		{
-			name: "predicate error",
-			pod:  &v1.Pod{Spec: volState},
+			name: "pv not found",
```

Comment:
`predicate error` is removed, and add two new tests which will return internal errors